### ISSUE TITLE
0.14.0: two tests started to fail while building debian pkg

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5402]
```
======================================================================
ERROR: datalad.support.tests.test_parallel.test_creatsubdatasets
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/tests/utils.py", line 752, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/tests/test_parallel.py", line 127, in test_creatsubdatasets
    assert_raises(IncompleteResultsError, list, ProducerConsumer(paths[::-1], create_, jobs=5))
  File "/usr/lib/python3.9/unittest/case.py", line 733, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/lib/python3.9/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 272, in __iter__
    yield from self._iter_threads(jobs)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 424, in _iter_threads
    self.shutdown(force=True, exception=self._producer_exception or interrupted_by_exception)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 237, in shutdown
    raise exception
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 408, in _iter_threads
    done_useful |= self._pop_done_futures(lgr)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 465, in _pop_done_futures
    raise exception
  File "/usr/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/parallel.py", line 336, in consumer_worker
    for r in res:
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/interface/utils.py", line 389, in generator_func
    for r in _process_results(
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/interface/utils.py", line 557, in _process_results
    for res in results:
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/core/local/create.py", line 396, in __call__
    tbrepo = AnnexRepo(
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/repo.py", line 154, in __call__
    instance = type.__call__(cls, *new_args, **new_kwargs)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/annexrepo.py", line 191, in __init__
    super(AnnexRepo, self).__init__(
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/gitrepo.py", line 940, in __init__
    self._create_empty_repo(path, create_sanity_checks, **git_opts)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/gitrepo.py", line 979, in _create_empty_repo
    os.makedirs(path)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/tmp/datalad_temp_test_creatsubdatasets8e80gels/subds1'
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using selector: EpollSelector
.... many more of the above
```

```
======================================================================
FAIL: datalad.support.tests.test_parallel.test_stalling
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/tests/utils.py", line 259, in _wrap_skip_if
    return func(*args, **kwargs)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/support/tests/test_parallel.py", line 256, in test_stalling
    raise AssertionError("Future has not finished in 5x time")
AssertionError: Future has not finished in 5x time
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using selector: EpollSelector
--------------------- >> end captured logging << ---------------------
```

```
root@smaug:/tmp/buildd/datalad-0.14.0# python3 --version
Python 3.9.1+
root@smaug:/tmp/buildd/datalad-0.14.0# git --version
git version 2.30.0
root@smaug:/tmp/buildd/datalad-0.14.0# git annex version
git-annex version: 8.20201127
...
```

yet to investigate

**Comment by yarikoptic** [comment]
damn, on rerunning `debian/rules binary` I did not get two above (the parallel one might be just flaky -- box was quite busy) but got 

```
FAIL: datalad.tests.test_log.test_filters
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/tests/test_log.py", line 133, in test_filters
    check_filters('datalad1')
  File "/build/datalad-0.14.0/.pybuild/cpython3_3.9_datalad/build/datalad/tests/test_log.py", line 125, in check_filters
    assert 'log3' not in cml.out
AssertionError:
```

grrr....

**Comment by kyleam** [comment]
> datalad.tests.test_log.test_filters

I see this one on my end with `--nocapture`.  Proposed a fix in gh-5407.

**Comment by yarikoptic** [comment]
thought to test @kyleam 's fix in #5407 but damn build passed this time just fine across the board, so did not get a chance to get to that "build environment" in which I produced the test_filters failure :-/  

meanwhile - uploaded those built 0.14.0 debian packages (well -- source package to debian proper).

So both test_parallel tests in original report are "flaky" under heavy load, and `test_creatsubdatasets` smells like some race condition which was not yet identified.

**Comment by yarikoptic** [comment]
I will just let it RiP until we awaken it from the dead
